t/t1091-sparse-checkout-builtin.sh

```diff
@@ -227,5 +227,16 @@ test_expect_success 'cone mode: init and set' '
 	test_cmp expect dir
 '
 
+test_expect_success 'cone mode: set with nested folders' '
+	git -C repo sparse-checkout set deep deep/deeper1 2>err &&
+	test_line_count = 0 err &&
+	cat >expect <<-EOF &&
+		/*
+		!/*/
+		/deep/
+	EOF
+	test_cmp repo/.git/info/sparse-checkout expect
```

Comment:
Do we need a test that has deeper folders like `a` and `a/b/c/d`?  

Also in looking at the code I don't think this matters but maybe a test the has the deep folder first before the parent so it passes `a/b/c a`?

Reply:
Since we are adding to a hashmap, the order does not matter. This is a white-box detail, as it is reasonable that an implementation _would_ care about the order, so we could repeat the test in the other order if we thought it mattered.